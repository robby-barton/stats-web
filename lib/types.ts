/* istanbul ignore file */

export type YearRanks = {
	weeks: number;
	postseason: boolean;
};

export type AvailRanks = {
	[index: string]: YearRanks;
};

export type Team = {
	team_id: number;
	name: string;
	logo: string;
	logo_dark: string;
};

export type AvailTeams = {
	[index: string]: Team;
};

export type Rank = {
	team: Team;
	conf: string;
	record: string;
	srs_rank: number;
	sos_rank: number;
	final_raw: number;
	final_rank: number;
};

export type TeamRank = {
	team: Team;
	final_rank: number;
	year: number;
	week: string;
	postseason: number;
};

export type TeamGames = {
	team: Team;
	sun: number;
	mon: number;
	tue: number;
	wed: number;
	thu: number;
	fri: number;
	sat: number;
	total: number;
};

export type RankingPath = {
	division: string;
	year: string;
	week: string;
};

export type TeamPath = {
	team: string;
};

export type ChartPoint = {
	week: string;
	rank: number;
	fillLevel: number;
};
