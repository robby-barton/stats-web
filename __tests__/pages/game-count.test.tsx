import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { TeamGames } from "@lib/types";
import GameCount from "@pages/game-count";

describe("Game Count page", () => {
	it("renders correctly", async () => {
		const games: TeamGames[] = [
			{
				name: "South Carolina",
				team_id: 2579,
				sun: 1,
				mon: 2,
				tue: 3,
				wed: 4,
				thu: 5,
				fri: 6,
				sat: 7,
				total: 8,
			},
		];
		const { baseElement } = render(<GameCount games={games} />);
		await screen.findByTitle(/light mode/i);

		expect(baseElement).toMatchSnapshot();
	});
});
