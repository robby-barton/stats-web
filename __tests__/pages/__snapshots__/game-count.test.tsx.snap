// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Count page renders correctly 1`] = `
<body
  data-theme="undefined"
>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div
          class="headerTitle"
        >
          <a
            href="/"
          >
            Robby's Ranking
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/game-count"
          >
            <span
              class="headingMd"
            >
              Game Count
            </span>
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/teams"
          >
            <span
              class="headingMd"
            >
              Teams
            </span>
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/about"
          >
            <span
              class="headingMd"
            >
              About
            </span>
          </a>
        </div>
        <div
          class="headerToggle"
        >
          <button
            aria-label="Change to light mode"
            class="css-1f6jnp0"
            title="Change to light mode"
            type="button"
          >
            <span
              class="css-194expn"
            />
            <span>
              üåô
            </span>
            <span>
              ‚òÄÔ∏è
            </span>
          </button>
        </div>
      </div>
      <div
        class="contentWrap"
      >
        <div>
          <div
            class="inputArea"
          >
            <div
              class="searchDiv"
            >
              <input
                class="searchInput"
                placeholder="Search Teams"
                type="search"
              />
            </div>
          </div>
          <table>
            <thead>
              <tr>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Team
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Sun
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Mon
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Tue
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Wed
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Thu
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Fri
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Sat
                  </div>
                </th>
                <th
                  colspan="1"
                >
                  <div
                    class="cursor-pointer select-none"
                  >
                    Total
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  South Carolina
                </td>
                <td>
                  1
                </td>
                <td>
                  2
                </td>
                <td>
                  3
                </td>
                <td>
                  4
                </td>
                <td>
                  5
                </td>
                <td>
                  6
                </td>
                <td>
                  7
                </td>
                <td>
                  8
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="footer"
      >
        <a
          href="https://github.com/robby-barton/stats-web"
        >
          &lt;/&gt; - Site source
        </a>
        <br />
        <a
          href="https://github.com/robby-barton/stats-go"
        >
          &lt;/&gt; - Ranking source
        </a>
      </div>
    </div>
  </div>
</body>
`;
